src/utils/extensionsUtils.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2019 Red Hat, Inc. and others.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { QExtension, APIExtension } from '../definitions/extensionInterfaces';
+
+export function convertToQExtensions(extensions: APIExtension[]): QExtension[] {
+  return extensions.map((extension) => {
+    const semicolon: number = extension.id.indexOf(':');
```

Comment:
You need to ensure there's a colon. If not, then assume the default groupId is io.quarkus. Need to check for empty value.
Need to check the extensions are not the old version of extensions (i.e. before this PR)

Reply:
Can we assume that the extension Ids from the endpoint, always contains a semicolon?

Reply:
Doesn't hurt to check